webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala

```diff
@@ -263,49 +302,75 @@ class CachedNodeChangesServiceImpl(
    * background processes where throughout is more
    * important than responsiveness.
    */
-  def update(changes: Seq[ResultRepairedReport]): Box[Unit] = {
+  def update(lowestId: Long, highestId: Long): Box[Unit] = {
     val enabled = computeChangeEnabled().getOrElse(true) // always true by default
     if(enabled) {
-      this.synchronized {
-        logger.debug(s"NodeChanges cache updating with ${changes.size} new changes...")
-        if(changes.isEmpty) {
-          Full(())
-        } else {
-          for {
-            existing <- initCache()
-          } yield {
-            val intervals = changeService.getCurrentValidIntervals(None)
-            val newChanges = changes.groupBy(_.ruleId).mapValues { ch =>
-              ( for {
-                 interval <- intervals
-              } yield {
-                (interval, ch.filter(interval contains _.executionTimestamp).size )
-              } ).toMap
-            }
-            logger.debug("NodeChanges cache updated")
-            cache = Some(merge(intervals, existing, newChanges))
-            logger.trace("NodeChanges cache content: " + cacheToLog(cache.get))
-            ()
-          }
-        }
-      }
+      QueuedChanges.queue.offer((lowestId, highestId))
+      Full(())
     } else {
       logger.warn(s"Not updating changes by rule - disabled by configuration setting 'rudder_compute_changes' (set by REST API)")
       Full(())
     }
   }
 
+  /*
+   * Get changes and update cache accordingly
+   */
+  protected def updateCache(lowestId: Long, highestId: Long): IOResult[Unit] = {
+    (for {
+      time1   <- currentTimeMillis
+      //
+      // actually, in syncUpdate we check if the cache was inited or not.
+      // when the cache is not, we do it - but what was the point to get
+      // changes here in that case ?
+      //
```

Comment:
I think the cache should keep the highestId it knows (either after init or after runs), and that we should get changes in `[lowestId, highestId]` interval, but in `[max(lowestId, cacheHighestId), highestId]`

Reply:
it is now done